### PR TITLE
[CSM] Add `get_wielded_item`

### PR DESCRIPTION
**Pull request by red-001**
I checked most of the itemstack methods and they seem to work as expected.

### CODE REVIEW

File: doc/client_lua_api.txt [hunk 1]
```diff
@@ -718,6 +718,9 @@ Call these functions only at load time!
     * Call the function `func` after `time` seconds, may be fractional
     * Optional: Variable number of arguments that are passed to `func`
 
+### Player
+* `minetest.get_wielded_item()`
+    * Returns the itemstack the local player is holding
```
Comment (nerzhul) [inline doc/client_lua_api.txt]: missing empty line

File: src/script/lua_api/l_client.h [hunk 2]
```diff
@@ -48,6 +48,12 @@ class ModApiClient:
 	// set_last_run_mod(modname)
 	static int l_set_last_run_mod(lua_State *L);
 
+
+	// get_wielded_item()
+	static int l_get_wielded_item(lua_State *L);
+
+
```
Comment (nerzhul) [inline src/script/lua_api/l_client.h]: too many empty lines before & after

File: src/script/lua_api/l_client.cpp [hunk 3]
```diff
@@ -97,6 +98,25 @@ int ModApiClient::l_gettext(lua_State *L)
...
+	Client *client = getClient(L);
+
+	Inventory local_inventory(client->idef());
+	client->getLocalInventory(local_inventory);
+
+	InventoryList *mlist = local_inventory.getList("main");
+
+	if (mlist && client->getPlayerItem() < mlist->getSize()) {
+		LuaItemStack::create(L, mlist->getItem(client->getPlayerItem()));
+	}
```
Comment (rubenwardy) [inline src/script/lua_api/l_client.cpp]: `} else {`

**Comment by sfan5** [comment]
What happens if that item stack is modified? nothing?

**Comment by red-001** [comment]
@sfan5 nothing as there is no function to set the current wielded item

File: src/script/lua_api/l_env.cpp [hunk 4]
```diff
@@ -472,7 +472,7 @@ int ModApiEnvMod::l_add_item(lua_State *L)
 	// pos
 	//v3f pos = checkFloatPos(L, 1);
 	// item
-	ItemStack item = read_item(L, 2,getServer(L));
+	ItemStack item = read_item(L, 2,getServer(L)->idef());
```
Comment (SmallJoker) [inline src/script/lua_api/l_env.cpp]: Space between `2,` and `getServer`

**Comment by nerzhul** [comment]
Thanks, merged to CSM branch

**Pull request merged**